README.md

```diff
@@ -13,3 +14,38 @@
   complexity suggests a different time.
 
 Add your answers to this markdown file.
+
+## My Response
+
+1.
+The first is that when we are analyzing an algorithm we ignore any constants that might arise, this can change things significantly.  On a lower level, $x^2$ and $4x^2$ are not the same function even though the would have the same complexity, $4x^2$ increases faster than its parent function.
+
+The second is that the complexity of an algorithm doesn't take into account the hardware of the machine running the program.  It could run faster or slower based on the resources available.
+
+Cade brought up this one while I was talking to him in lab.  The bounds of our complexity can come into question when we start looking at Big Theta complexity.  For instance the $\Theta$ complexity of a program might be $nlogn$ but its upper and lower are $2^n$ and $n$ respectively, which are two very different functions that leave a lot of room for movement between them.
+
+2.
+According to Wikipedia the average time complexity of searching for a value in a binary search tree is $log(n)$, with this we can compute the approximate time for the program to finish.
+
+first we need the values of $log(1000)$ and $log(10000)$ these we can then put into a ratio as to the difference between 1000 elements and 10000 elements
+
+$log(1000) = 3$
+
+$log(10000) = 4$
+
+Now we can build an equation similar to how chemistry builds their unit conversions $5seconds/3searches * 4searches / 1$ this would cancel out our searches and leave us with $5seconds / 3 * 4 / 1$ which further simplifies to $1.6667seconds * 4 \approx 6.7seconds$.
+
+3.
+The algorithm might not be efficient enough to handle these higher element trees, in other words, it might not have a good enough time complexity to make finding elements viable requiring further bug fixing.
+
+Like with the second point I provided above, these two different algorithms might have been run on two different computers or something was going on in the background that was taking up avaliable resources, then the two different runs could still have the same asymptotic complexity but they wouldn't necessarily have predictable outcomes.
+
+The value we are looking for might have been omitted from the graph itself, this would mean that the algorithm would have to run through every single potential element until it determined that the requested value was absent.
```

Comment:
Would that account for such a large difference?

Reply:
For that first one yes I do believe so, because if there is an older computer running it then it can take longer to work through that process than a modern machine.

For the second I still think it holds up, because if it can't find the value in the shortest position won't it run through the rest of the tree to try and find it?

Reply:
Yes, but the difference in tree height is only 3 (10 for the smaller input, 13 for the larger). How would that cause such a large difference?

Reply:
Sounds like this might go in the direction of a reason, but you'll have to be more specific.

Reply:
Well strings can get quite long compared to integers, and if it was having to check each string as it went and with enough strings this could really slow down the delivery time of our search.  I guess a better way to put it would be 17 is a lot easier to find than "Laramie, Wyoming"

Reply:
I believe so

Reply:
Ok, but you'll have to convince me as well :)